{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "98608277",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2024-03-18 15:39:40\n",
      "2024-03-18 15:39:47\n",
      "2024-03-18 15:39:54\n",
      "2024-03-18 15:40:01\n",
      "2024-03-18 15:40:09\n",
      "2024-03-18 15:40:17\n",
      "2024-03-18 15:40:24\n",
      "2024-03-18 15:40:32\n",
      "2024-03-18 15:40:40\n",
      "2024-03-18 15:40:47\n",
      "2024-03-18 15:40:55\n",
      "2024-03-18 15:41:02\n",
      "2024-03-18 15:41:10\n",
      "2024-03-18 15:41:17\n",
      "2024-03-18 15:41:25\n",
      "2024-03-18 15:41:33\n",
      "2024-03-18 15:41:41\n",
      "2024-03-18 15:41:48\n",
      "2024-03-18 15:41:56\n",
      "2024-03-18 15:42:03\n",
      "2024-03-18 15:42:11\n",
      "2024-03-18 15:42:20\n",
      "2024-03-18 15:42:28\n",
      "2024-03-18 15:42:36\n",
      "2024-03-18 15:42:44\n",
      "2024-03-18 15:42:52\n",
      "2024-03-18 15:43:00\n",
      "2024-03-18 15:43:10\n",
      "2024-03-18 15:43:18\n",
      "2024-03-18 15:43:27\n",
      "2024-03-18 15:43:35\n",
      "2024-03-18 15:43:43\n",
      "2024-03-18 15:43:51\n",
      "2024-03-18 15:44:00\n",
      "2024-03-18 15:44:08\n",
      "2024-03-18 15:44:16\n",
      "2024-03-18 15:44:25\n",
      "2024-03-18 15:44:33\n",
      "2024-03-18 15:44:41\n",
      "2024-03-18 15:44:50\n",
      "2024-03-18 15:44:58\n",
      "2024-03-18 15:45:06\n",
      "2024-03-18 15:45:14\n",
      "2024-03-18 15:45:23\n",
      "2024-03-18 15:45:32\n",
      "2024-03-18 15:45:41\n",
      "2024-03-18 15:45:49\n",
      "2024-03-18 15:45:57\n",
      "2024-03-18 15:46:06\n",
      "2024-03-18 15:46:15\n",
      "2024-03-18 15:46:24\n",
      "2024-03-18 15:46:34\n",
      "2024-03-18 15:46:42\n",
      "2024-03-18 15:46:51\n",
      "2024-03-18 15:47:00\n",
      "2024-03-18 15:47:10\n",
      "2024-03-18 15:47:18\n",
      "2024-03-18 15:47:27\n",
      "2024-03-18 15:47:36\n",
      "2024-03-18 15:47:45\n",
      "2024-03-18 15:47:54\n",
      "2024-03-18 15:48:03\n",
      "2024-03-18 15:48:12\n",
      "2024-03-18 15:48:21\n",
      "2024-03-18 15:48:31\n",
      "2024-03-18 15:48:40\n",
      "2024-03-18 15:48:49\n",
      "2024-03-18 15:48:58\n",
      "2024-03-18 15:49:07\n",
      "2024-03-18 15:49:16\n",
      "2024-03-18 15:49:25\n",
      "2024-03-18 15:49:34\n",
      "2024-03-18 15:49:44\n",
      "2024-03-18 15:49:53\n",
      "2024-03-18 15:50:03\n",
      "2024-03-18 15:50:13\n",
      "2024-03-18 15:50:22\n",
      "2024-03-18 15:50:31\n",
      "2024-03-18 15:50:40\n",
      "2024-03-18 15:50:50\n",
      "2024-03-18 15:51:00\n",
      "2024-03-18 15:51:10\n",
      "2024-03-18 15:51:19\n",
      "2024-03-18 15:51:28\n",
      "2024-03-18 15:51:38\n",
      "2024-03-18 15:51:48\n",
      "2024-03-18 15:51:58\n",
      "2024-03-18 15:52:08\n",
      "2024-03-18 15:52:18\n",
      "2024-03-18 15:52:28\n",
      "2024-03-18 15:52:38\n",
      "2024-03-18 15:52:48\n",
      "2024-03-18 15:52:58\n",
      "2024-03-18 15:53:07\n",
      "2024-03-18 15:53:17\n",
      "2024-03-18 15:53:27\n",
      "2024-03-18 15:53:37\n",
      "2024-03-18 15:53:48\n",
      "2024-03-18 15:53:57\n",
      "2024-03-18 15:54:08\n",
      "2024-03-18 15:54:18\n",
      "2024-03-18 15:54:28\n",
      "2024-03-18 15:54:38\n",
      "2024-03-18 15:54:48\n",
      "2024-03-18 15:54:58\n",
      "2024-03-18 15:55:08\n",
      "2024-03-18 15:55:18\n",
      "2024-03-18 15:55:28\n",
      "2024-03-18 15:55:39\n",
      "2024-03-18 15:55:49\n",
      "2024-03-18 15:55:59\n",
      "2024-03-18 15:56:10\n",
      "2024-03-18 15:56:20\n",
      "2024-03-18 15:56:31\n",
      "2024-03-18 15:56:42\n",
      "2024-03-18 15:56:52\n",
      "2024-03-18 15:57:03\n",
      "2024-03-18 15:57:14\n",
      "2024-03-18 15:57:25\n",
      "2024-03-18 15:57:36\n",
      "2024-03-18 15:57:46\n",
      "2024-03-18 15:57:57\n",
      "2024-03-18 15:58:08\n",
      "2024-03-18 15:58:19\n",
      "2024-03-18 15:58:29\n",
      "2024-03-18 15:58:41\n",
      "2024-03-18 15:58:52\n",
      "2024-03-18 15:59:03\n",
      "2024-03-18 15:59:14\n",
      "2024-03-18 15:59:25\n",
      "2024-03-18 15:59:37\n",
      "2024-03-18 15:59:48\n",
      "2024-03-18 15:59:59\n",
      "2024-03-18 16:00:10\n",
      "2024-03-18 16:00:21\n",
      "2024-03-18 16:00:32\n",
      "2024-03-18 16:00:44\n",
      "2024-03-18 16:00:55\n",
      "2024-03-18 16:01:06\n",
      "2024-03-18 16:01:17\n",
      "2024-03-18 16:01:29\n",
      "2024-03-18 16:01:40\n",
      "2024-03-18 16:01:52\n",
      "2024-03-18 16:02:04\n",
      "2024-03-18 16:02:15\n",
      "2024-03-18 16:02:26\n",
      "2024-03-18 16:02:39\n",
      "2024-03-18 16:02:50\n",
      "2024-03-18 16:03:01\n",
      "2024-03-18 16:03:13\n",
      "2024-03-18 16:03:25\n",
      "2024-03-18 16:03:36\n",
      "2024-03-18 16:03:49\n",
      "2024-03-18 16:04:01\n",
      "2024-03-18 16:04:13\n",
      "2024-03-18 16:04:24\n",
      "2024-03-18 16:04:36\n",
      "2024-03-18 16:04:48\n",
      "2024-03-18 16:05:01\n",
      "2024-03-18 16:05:13\n",
      "2024-03-18 16:05:25\n",
      "2024-03-18 16:05:37\n",
      "2024-03-18 16:05:49\n",
      "2024-03-18 16:06:01\n",
      "2024-03-18 16:06:13\n",
      "2024-03-18 16:06:25\n",
      "2024-03-18 16:06:37\n",
      "2024-03-18 16:06:49\n",
      "2024-03-18 16:07:01\n",
      "2024-03-18 16:07:13\n",
      "2024-03-18 16:07:25\n",
      "2024-03-18 16:07:38\n",
      "2024-03-18 16:07:50\n",
      "2024-03-18 16:08:03\n",
      "2024-03-18 16:08:16\n",
      "2024-03-18 16:08:29\n",
      "2024-03-18 16:08:41\n",
      "2024-03-18 16:08:53\n",
      "2024-03-18 16:09:06\n",
      "2024-03-18 16:09:18\n",
      "2024-03-18 16:09:31\n",
      "2024-03-18 16:09:43\n",
      "2024-03-18 16:09:56\n",
      "2024-03-18 16:10:08\n",
      "2024-03-18 16:10:21\n",
      "2024-03-18 16:10:33\n",
      "2024-03-18 16:10:46\n",
      "2024-03-18 16:10:59\n"
     ]
    }
   ],
   "source": [
    "#pip install mysql-connector\n",
    "import mysql.connector as my_sql\n",
    "my_db = my_sql.connect(\n",
    "    host=\"localhost\",\n",
    "    user=\"Arduino\",\n",
    "    passwd=\"ArduinoGetStarted.com\",\n",
    "    database=\"vital_signs\")\n",
    "my_cursor = my_db.cursor()\n",
    "    \n",
    "# Execute the SQL query to fetch real-time data\n",
    "my_cursor.execute(\"SELECT * FROM radar_sensor\")    \n",
    "data = my_cursor.fetchall()\n",
    "    \n",
    "for row in data:\n",
    "    print(row[2])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "a2cc755e-2359-44a4-af59-c8b6f99cbd52",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Public\\Documents\\Wondershare\\CreatorTemp\\ipykernel_3272\\1880136665.py:43: UserWarning: frames=None which we can infer the length of, did not pass an explicit *save_count* and passed cache_frame_data=True.  To avoid a possibly unbounded cache, frame data caching has been disabled. To suppress this warning either pass `cache_frame_data=False` or `save_count=MAX_FRAMES`.\n",
      "  ani = animation.FuncAnimation(fig, animate, interval=1000)\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\mysql\\connector\\network.py\", line 509, in open_connection\n",
      "    self.sock.connect(sockaddr)\n",
      "ConnectionRefusedError: [WinError 10061] No connection could be made because the target machine actively refused it\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\tkinter\\__init__.py\", line 1948, in __call__\n",
      "    return self.func(*args)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\tkinter\\__init__.py\", line 861, in callit\n",
      "    func(*args)\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\matplotlib\\backends\\_backend_tk.py\", line 145, in _on_timer\n",
      "    super()._on_timer()\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\matplotlib\\backend_bases.py\", line 1233, in _on_timer\n",
      "    ret = func(*args, **kwargs)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\matplotlib\\animation.py\", line 1427, in _step\n",
      "    still_going = super()._step(*args)\n",
      "                  ^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\matplotlib\\animation.py\", line 1121, in _step\n",
      "    self._draw_next_frame(framedata, self._blit)\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\matplotlib\\animation.py\", line 1140, in _draw_next_frame\n",
      "    self._draw_frame(framedata)\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\matplotlib\\animation.py\", line 1768, in _draw_frame\n",
      "    self._drawn_artists = self._func(framedata, *self._args)\n",
      "                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\Public\\Documents\\Wondershare\\CreatorTemp\\ipykernel_3272\\1880136665.py\", line 24, in animate\n",
      "    my_db = my_sql.connect(\n",
      "            ^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\mysql\\connector\\__init__.py\", line 179, in connect\n",
      "    return MySQLConnection(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\mysql\\connector\\connection.py\", line 95, in __init__\n",
      "    self.connect(**kwargs)\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\mysql\\connector\\abstracts.py\", line 716, in connect\n",
      "    self._open_connection()\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\mysql\\connector\\connection.py\", line 206, in _open_connection\n",
      "    self._socket.open_connection()\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\mysql\\connector\\network.py\", line 511, in open_connection\n",
      "    raise errors.InterfaceError(\n",
      "mysql.connector.errors.InterfaceError: 2003: Can't connect to MySQL server on 'localhost:3306' (10061 No connection could be made because the target machine actively refused it)\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\mysql\\connector\\network.py\", line 140, in send_plain\n",
      "    self.sock.sendall(packet)\n",
      "ConnectionResetError: [WinError 10054] An existing connection was forcibly closed by the remote host\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\tkinter\\__init__.py\", line 1948, in __call__\n",
      "    return self.func(*args)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\tkinter\\__init__.py\", line 861, in callit\n",
      "    func(*args)\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\matplotlib\\backends\\_backend_tk.py\", line 145, in _on_timer\n",
      "    super()._on_timer()\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\matplotlib\\backend_bases.py\", line 1233, in _on_timer\n",
      "    ret = func(*args, **kwargs)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\matplotlib\\animation.py\", line 1427, in _step\n",
      "    still_going = super()._step(*args)\n",
      "                  ^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\matplotlib\\animation.py\", line 1121, in _step\n",
      "    self._draw_next_frame(framedata, self._blit)\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\matplotlib\\animation.py\", line 1140, in _draw_next_frame\n",
      "    self._draw_frame(framedata)\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\matplotlib\\animation.py\", line 1768, in _draw_frame\n",
      "    self._drawn_artists = self._func(framedata, *self._args)\n",
      "                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\Public\\Documents\\Wondershare\\CreatorTemp\\ipykernel_3272\\863482925.py\", line 29, in animate\n",
      "    my_cursor.execute(\"SELECT * FROM radar_sensor\")\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\mysql\\connector\\cursor.py\", line 551, in execute\n",
      "    self._handle_result(self._connection.cmd_query(stmt))\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\mysql\\connector\\connection.py\", line 490, in cmd_query\n",
      "    result = self._handle_result(self._send_cmd(ServerCmd.QUERY, query))\n",
      "                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\mysql\\connector\\connection.py\", line 259, in _send_cmd\n",
      "    self._socket.send(\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\mysql\\connector\\network.py\", line 142, in send_plain\n",
      "    raise errors.OperationalError(\n",
      "mysql.connector.errors.OperationalError: 2055: Lost connection to MySQL server at 'localhost:3306', system error: 10054 An existing connection was forcibly closed by the remote host\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\mysql\\connector\\network.py\", line 140, in send_plain\n",
      "    self.sock.sendall(packet)\n",
      "ConnectionResetError: [WinError 10054] An existing connection was forcibly closed by the remote host\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\tkinter\\__init__.py\", line 1948, in __call__\n",
      "    return self.func(*args)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\tkinter\\__init__.py\", line 861, in callit\n",
      "    func(*args)\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\matplotlib\\backends\\_backend_tk.py\", line 145, in _on_timer\n",
      "    super()._on_timer()\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\matplotlib\\backend_bases.py\", line 1233, in _on_timer\n",
      "    ret = func(*args, **kwargs)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\matplotlib\\animation.py\", line 1427, in _step\n",
      "    still_going = super()._step(*args)\n",
      "                  ^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\matplotlib\\animation.py\", line 1121, in _step\n",
      "    self._draw_next_frame(framedata, self._blit)\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\matplotlib\\animation.py\", line 1140, in _draw_next_frame\n",
      "    self._draw_frame(framedata)\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\matplotlib\\animation.py\", line 1768, in _draw_frame\n",
      "    self._drawn_artists = self._func(framedata, *self._args)\n",
      "                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\Public\\Documents\\Wondershare\\CreatorTemp\\ipykernel_3272\\1423734333.py\", line 29, in animate\n",
      "    my_cursor.execute(\"SELECT * FROM radar_sensor\")\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\mysql\\connector\\cursor.py\", line 551, in execute\n",
      "    self._handle_result(self._connection.cmd_query(stmt))\n",
      "                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\mysql\\connector\\connection.py\", line 490, in cmd_query\n",
      "    result = self._handle_result(self._send_cmd(ServerCmd.QUERY, query))\n",
      "                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\mysql\\connector\\connection.py\", line 259, in _send_cmd\n",
      "    self._socket.send(\n",
      "  File \"C:\\Users\\natha\\anaconda3\\Lib\\site-packages\\mysql\\connector\\network.py\", line 142, in send_plain\n",
      "    raise errors.OperationalError(\n",
      "mysql.connector.errors.OperationalError: 2055: Lost connection to MySQL server at 'localhost:3306', system error: 10054 An existing connection was forcibly closed by the remote host\n"
     ]
    }
   ],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "%matplotlib tk\n",
    "import matplotlib.animation as animation\n",
    "import time\n",
    "#libraries we need:\n",
    "import mysql.connector as my_sql\n",
    "import matplotlib.animation as animation \n",
    "import matplotlib.pyplot as plt \n",
    "import numpy as np \n",
    "global data\n",
    "\n",
    "fig = plt.figure()\n",
    "ax1 = fig.add_subplot(1,1,1)\n",
    "\n",
    "\n",
    "    \n",
    "# Execute the SQL query to fetch real-time data\n",
    "#my_cursor.execute(\"SELECT * FROM radar_sensor\")    \n",
    "#data = my_cursor.fetchall()\n",
    "\n",
    "def animate(i):\n",
    "    #pullData = open(\"sampleText.txt\",\"r\").read()\n",
    "    #dataArray = pullData.split('\\n')\n",
    "    my_db = my_sql.connect(\n",
    "        host=\"localhost\",\n",
    "        user=\"Arduino\",\n",
    "        passwd=\"ArduinoGetStarted.com\",\n",
    "        database=\"vital_signs\")\n",
    "    my_cursor = my_db.cursor()\n",
    "    my_cursor.execute(\"SELECT * FROM radar_sensor\")    \n",
    "    data = my_cursor.fetchall()\n",
    "    xar = []\n",
    "    yar = []\n",
    "    for row in data:\n",
    "        xar.append(row[0]) #timestamp - id actually\n",
    "        #print(row)\n",
    "        yar.append(row[4]) #heartrate; row 4 = breathing rate\n",
    "    ax1.clear()\n",
    "    ax1.plot(xar,yar)\n",
    "    #ax1.plot(yar)\n",
    "\n",
    "#ani = animation.FuncAnimation(fig, animate, fargs=(xs, ys), interval=1000)\n",
    "ani = animation.FuncAnimation(fig, animate, interval=1000)\n",
    "#plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9054e1dd",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "14bea899-4947-446b-ab7a-333fdc8540c5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f01646c9-2683-4974-98eb-ecc0b052b4fe",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
